functions/shiritori/index.js

```diff
@@ -14,116 +14,104 @@
 
 'use strict'
 
-const kuroshiro = require('kuroshiro')
+const Kuroshiro = require('kuroshiro')
+const KuromojiAnalyzer = require('kuroshiro-analyzer-kuromoji')
 const wanakana = require('wanakana')
 
 const smallKanas = 'ぁぃぅぇぉゃゅょ'
-let kuroshiroLoaded = false
 
 // 辞書のローディング。
-exports.loaded = new Promise((resolve, reject) => {
-  kuroshiro.init(function (err) {
-    if (err) {
-      reject(err)
-      return
-    }
-    kuroshiroLoaded = true
-    resolve()
-  })
-})
-
-// しりとりのルールのチェック。
-exports.check = (word, chain) => {
-  if ((chain === undefined) || (chain.length === 0)) {
-    return true
-  }
-
-  if (kuroshiroLoaded) {
-    word = kuroshiro.toHiragana(word)
-    chain = chain.map((e) => kuroshiro.toHiragana(e))
-  }
+const kuroshiro = new Kuroshiro()
+const loaded = kuroshiro.init(new KuromojiAnalyzer())
 
-  // 漢字からひらがなにする。
-  const wordHira = wanakana.toHiragana(word)
-  const chainHira = chain.map(wanakana.toHiragana)
+class Win extends Error {}
+exports.Win = Win
+class Bad extends Error {}
+exports.Bad = Bad
 
-  // 使った名詞をチェックする。
-  if (chainHira.indexOf(wordHira, 0) !== -1) {
-    return false
-  }
+// しりとりのルールのチェック。
+exports.check = (word, chain) => loaded.then(() => {
+  return Promise.all([
+    kuroshiro.convert(word, { to: 'hiragana' }),
+    ...chain.map(e => kuroshiro.convert(e, { to: 'hiragana' }))
+  ]).then(([word, ...chain]) => {
+    // 漢字からひらがなにする。
+    const wordHira = wanakana.toHiragana(word)
+    if (wordHira[wordHira.length - 1] === 'ん') {
+      throw new Bad('ends with ん')
+    }
 
-  // しりとりの最初の文字をチェックする。
-  const validKanas = exports.kanas(chain[0])
-  for (const k of validKanas) {
-    const begin = wordHira.slice(0, k.length)
-    if (begin === k) {
+    if ((chain === undefined) || (chain.length === 0)) {
       return true
     }
-  }
-  return false
-}
 
-// 名詞からしりとりのひらがなを選ぶ。
-exports.kanas = (word) => new Set((() => {
-  if (kuroshiroLoaded) {
-    word = kuroshiro.toHiragana(word)
-  }
+    // 使った名詞をチェックする。
+    const chainHira = chain.map(wanakana.toHiragana)
+    if (chainHira.indexOf(wordHira, 0) !== -1) {
```

Comment:
You should be able to do `chainHira.includes(wordHira)`

Reply:
Done